notebooks/ch16/classification_precision_recall.md

```diff
@@ -0,0 +1,326 @@
+
+The precision of `words_list_model` is $ \frac{3}{3 + 4} \approx .43 $, whereas the precision of `ham_only` is 0 (do you see why?). This validates our expectation that `words_list_model` is a better classifier than `ham_only`, but there is still significant room to improve it.
+
+See [here]() for scikit-learn's precision function.
+
+### Recall
+
+**Recall** (also called **sensitivity**) measures the proportion of actual positive data that is classified as positive.
+
+$$ \text{Recall} = \frac{TP}{TP + FN} $$
+
+A model with high recall accurately recognizes aspects of positive data, whereas a model with low recall isn't sensitive enough to what distinguishes positive from negative data, and as a result leads to ample false negatives.
+
+The recall of `words_list_model` is $ \frac{3}{3 + 33} \approx .083$, whereas the recall of `ham_only` is 0 (do you see why?). This is further confirmation that `ham_only` is useless; however the abysmally low recall of `words_list_model` is a strong indication that we need to do much more EDA and feature engineering to improve the model.
+
+See [here]() for scikit-learn's recall function.
+
+### Precision-Recall Curve
+
+Both precision and recall are needed to thoroughly evaluate a model. Although the goal is to maximize both metrics, there is a tradeoff between increasing precision and increasing recall. Generally, increasing precision leads to a decrease in recall and vice versa (though there may be some cases where they move in the same direction). For instance, reducing the number of false negatives by training the model to recognize more features of the positive class can increase false positives on data for which the model is less certain of its classification.
+
+The **classification threshold** is a value that determines what class a data point is assigned to; points on opposite sides of the threshold are labelled with different classes. Recall that logistic regression outputs a probability that the data point belongs to the positive class. If the probability output on $x_1$ is .90 and the output on $x_2$ is .1, then we can be fairly confident that $x_1$ is spam and $x_2$ is ham. However, if the probability associated with $x_3$ is around .55, then we have less confidence and based on the classification threshold, $x_3$ may be labelled as spam in some cases and ham in others. Depending on the problem statement, false positives and false negatives may have different importance, so set the classification threshold based on which error type you want to minimize. For email classification, false positives result in important emails being filtered out, so they are worse than false negative, in which a spam email winds up in the inbox. In medical settings, however, false negatives in a diagnostic test can be much more consequential than false positives.
```

Comment:
- "they are worse than false negative" --> "they are worse than false negatives"
- While talking about whether we should prioritize FN/FP minimization, it's probably also a good idea to mention whether that corresponds to Precision or Recall
- labelled/labeled